csrc/ir/utils.cpp

```diff
@@ -1698,4 +1698,32 @@ std::vector<IterDomain*> getReachableIds(
   return dependent_ids;
 }
 
+std::vector<IterDomain*> propagateScatterAllocationDomain(
+    TensorView* scatter_out,
+    const std::vector<IterDomain*>& to_logical_domain) {
+  NVF_ERROR_EQ(
+      scatter_out->getLogicalDomain().size(),
+      to_logical_domain.size(),
+      "Mismatching tensor rank");
+  if (!scatter_out->hasAllocation()) {
+    return to_logical_domain;
```

Comment:
~nitpick: conceptually, should this return be moved before the assert?~

nvm, I see the assert above is to fill the gap of the lack of mapping. so it makes sense to still assert even though we are not replaying anything.